exercises/wordy/src/example/java/WordProblemSolver.java

```diff
@@ -0,0 +1,60 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class WordProblemSolver {
+
+    private static final String INTEGER_REGEX_STRING = "(-?\\d+)";
+
+    private static final String BINARY_OPERATION_REGEX_STRING
+        = "(plus|minus|multiplied by|divided by|raised to the power)";
+
+    private static final String VALID_QUESTION_REGEX_STRING =
```

Comment:
Splitting into components allows for the entire problem to be validated, as well as the individual pieces of information to be extracted for the actual calculations. For example, not checking the overall structure could allow a question like

> What is 3 nonsense words lorem ipsum plus 4?

to be considered valid. Still feels quite awkward though.

I initially tried writing a single regex string that would grab every piece of information required at once while simultaneously validating the entire problem structure, but then I ran into difficulties with [backtracking](http://www.regular-expressions.info/captureall.html).